### PR TITLE
Add GlobalPos versions of IWandable methods with fallback if dimensions are the same

### PR DESCRIPTION
**Pull request by Vonr**
Allows for cross-dimensional linking shenanigans.

Dominion Wand might be a bit broken, I've done some testing but the changes are janky.

### CODE REVIEW

File: src/main/java/com/hollingsworth/arsnouveau/api/item/IWandable.java [hunk 1]
```diff
@@ -3,34 +3,110 @@
...
      */
+    default void onFinishedConnectionFirst(@Nullable GlobalPos storedPos, @Nullable LivingEntity storedEntity, Player playerEntity) {
+        ResourceKey<Level> dim;
+        if (storedPos != null) {
+            dim = storedPos.dimension();
+        } else if (storedEntity != null) {
+            dim = storedEntity.level.dimension();
+        } else {
+            return;
+        }
```
Comment (Jarva) [inline src/main/java/com/hollingsworth/arsnouveau/api/item/IWandable.java:29]: Could this be moved into a private function on the interface? Saves repeating the same logic 4 times

File: src/main/java/com/hollingsworth/arsnouveau/common/items/DominionWand.java [hunk 2]
```diff
@@ -169,7 +172,7 @@ public void appendHoverText(@NotNull ItemStack stack, @NotNull TooltipContext wo
         if (data.storedPos().isEmpty()) {
             tooltip.add(Component.translatable("ars_nouveau.dominion_wand.no_location"));
         } else {
-            tooltip.add(Component.translatable("ars_nouveau.dominion_wand.position_stored", getPosString(data.getValidPos())));
+            tooltip.add(Component.translatable("ars_nouveau.dominion_wand.position_stored", getGlobalPosString(data.getValidPos())));
```
Comment (Jarva) [inline src/main/java/com/hollingsworth/arsnouveau/common/items/DominionWand.java:175]: Should getPosString be removed/deprecated? Is it used anywhere else?

**Review by Jarva — CHANGES_REQUESTED** [review]
Couple small tweaks for maintainability.

Could I also get a ping when this is merged? I imagine it'll break the AdvancedDominionWand

File: src/main/java/com/hollingsworth/arsnouveau/common/items/DominionWand.java (hunk 2 above)
Comment (Vonr) [reply src/main/java/com/hollingsworth/arsnouveau/common/items/DominionWand.java:175]: Yep, they are still used in RedstoneRelayTile and RelayTile.

File: src/main/java/com/hollingsworth/arsnouveau/api/item/IWandable.java (hunk 1 above)
Comment (Vonr) [reply src/main/java/com/hollingsworth/arsnouveau/api/item/IWandable.java]: Resolved

File: src/main/java/com/hollingsworth/arsnouveau/common/items/data/DominionWandData.java [hunk 3]
```diff
@@ -21,14 +22,14 @@ public DominionWandData(){
     public static final int NULL_ENTITY = -1;
 
     public static Codec<DominionWandData> CODEC = RecordCodecBuilder.create(instance -> instance.group(
-            BlockPos.CODEC.optionalFieldOf("pos").forGetter(DominionWandData::storedPos),
+            GlobalPos.CODEC.optionalFieldOf("pos").forGetter(DominionWandData::storedPos),
```
Comment (baileyholl) [inline src/main/java/com/hollingsworth/arsnouveau/common/items/data/DominionWandData.java:25]: Does this break existing worlds? If you save a value on the wand as Blockpos, change this to global and then reboot the world, does it still load with that item?

Comment (Vonr) [reply src/main/java/com/hollingsworth/arsnouveau/common/items/data/DominionWandData.java:25]: good question, will test later

Comment (Vonr) [reply src/main/java/com/hollingsworth/arsnouveau/common/items/data/DominionWandData.java:25]: not exactly sure what happens, but the world loads fine and the wand is treated as having no stored position. stored entity carries through though.

**Pull request merged**